dev/tools/controllerbuilder/pkg/gocode/ast.go

```diff
@@ -144,6 +144,10 @@ func (p *Package) inspect(packageName string, pkg *ast.Package) error {
 				}
 			case *ast.Ident:
 				// type alias
+			case *ast.InterfaceType:
```

Comment:
I see. So it is trying to parse 
https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/a1ffee0774b3a018301a2b7f0525e8402c0f5bd4/apis/refs/v1beta1/interface.go#L23

Mapper do not need to parse Go interface. I think we can always skip on *ast.InterfaceType.

Reply:
I'm wondering if we shall only load the packages that match the flag Service/Kind. I guess the loading of `./refs` may come from that the resource has references from `./refs`. In which case, the struct always contains `Name, Namespace, External` and no further dependencies. @jingyih WDYT?

Reply:
@acpana What happens if we pass `--api-dir $REPO_ROOT/apis/bigqueryanalyticshub` to the mapper generator?
(I assume you have the generated types under $REPO_ROOT/apis/bigqueryanalyticshub)

My understanding is that all the types for `bigqueryanalyticshub` should be under "$REPO_ROOT/apis/bigqueryanalyticshub", there is no need to parse other subfolders under "$REPO_ROOT/apis". Mapping functions for referenced fields are handled manually anyway.

Reply:
Thanks for confirmation! Let's skip "interface" for now. If we need it in future, we can always add it back!

Reply:
Thanks for the suggestion, @jingyih! I'd like to have the PR goes in so it can unblock others who run `dev/tasks/generate.sh` or the generate-mapper tool as well. Feel free to propose new approaches for mapper generation/update.